xmr-btc/src/happy_path.rs

```diff
@@ -0,0 +1,216 @@
+//! This module shows how a BTC/XMR atomic swap proceeds along the happy path.
+
+use crate::{alice, bitcoin, bob, monero, Message, ReceiveMessage, SendMessage};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use bitcoin_harness::Bitcoind;
+use futures::{
+    channel::{
+        mpsc,
+        mpsc::{Receiver, Sender},
+    },
+    SinkExt, StreamExt,
+};
+use monero_harness::Monero;
+use rand::rngs::OsRng;
+use std::convert::TryInto;
+use testcontainers::clients::Cli;
+
+const TEN_XMR: u64 = 10_000_000_000_000;
+
+pub async fn init_bitcoind(tc_client: &Cli) -> Bitcoind<'_> {
+    let bitcoind = Bitcoind::new(tc_client, "0.19.1").expect("failed to create bitcoind");
+    let _ = bitcoind.init(5).await;
+
+    bitcoind
+}
+
+/// Create two mock `Transport`s which mimic a peer to peer connection between
+/// two parties, allowing them to send and receive `thor::Message`s.
+pub fn make_transports() -> (Transport, Transport) {
```

Comment:
This looks similar to what @rishflab did in his PR in https://github.com/comit-network/xmr-btc-swap/pull/2 
see https://github.com/comit-network/xmr-btc-swap/blob/a5e9d12a6c6c489eccd6de6dd4ae85da38f2e60f/xmr-btc/tests/e2e.rs#L41-L60

Reply:
Yes except he puts two methods on a single trait - a single method per trait is cleaner. Also his traits are never used, writing these traits takes about 2 minutes of cut and pasting from `thor`. I do not see this as an issue.

Reply:
True, he uses a single trait. That can be refactored easily. 

It looks like the traits are used in the `next_state` function for Alice and Bob 

https://github.com/comit-network/xmr-btc-swap/blob/b6df715b1bdef81f20333145e1705cbdcf0e4fb8/xmr-btc/src/alice.rs#L34-L35

and 

https://github.com/comit-network/xmr-btc-swap/blob/b6df715b1bdef81f20333145e1705cbdcf0e4fb8/xmr-btc/src/bob.rs#L37-L41

Reply:
> Tthe traits are used in the next_state function for Alice and Bob

The call sites you reference are using the function parameter that is bound by the trait. This code is not using the implementation of the trait that we are discussing.